File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = defaultdict(int)
        for num in nums:
            map[num] += 1

        heap = []
        for n in map.keys():
            heapq.heappush(heap, (map[n], n))
            if len(heap) > k:
                heapq.heappop(heap)

        result = []
        for _ in range(k):
            result.append(heapq.heappop(heap)[1])
        

        return result

File: Data Structures & Algorithms/subarray-product-less-than-k/submission-0.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int l = 0, result = 0;
        long product = 1;

        for (int r = 0; r < nums.length; r++) {
            product *= nums[r];
            while (l <= r && product >= k) {
                product /= nums[l];
                l++;
            }
            result += r - l + 1;
        }
        return result;
    }
}